spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -139,15 +142,14 @@ public void executeTask(String taskName, Map<String, String> runtimeProperties,
 
 		taskDefinition = this.updateTaskProperties(taskDefinition);
 
-		Map<String, String> deploymentProperties = new HashMap<>(runtimeProperties);
-		URI uri = this.registry.find(String.format("task.%s",
-				taskDefinition.getRegisteredAppName()));
+		URI uri = this.registry.find(String.format("task.%s", taskDefinition.getRegisteredAppName()));
 		Resource resource = this.resourceLoader.getResource(uri.toString());
 
-		taskDefinition = qualifyProperties(taskDefinition, resource);
 
-		AppDeploymentRequest request = taskDefinition.createDeploymentRequest(resource,
-				deploymentProperties, runtimeParams);
+		Map<String, String> appDeploymentProperties = extractAppProperties(taskDefinition, taskDeploymentProperties);
+		Map<String, String> deployerDeploymentProperties = DeploymentPropertiesUtils.extractAndQualifyDeployerProperties(taskDeploymentProperties, taskDefinition.getName());
```

Comment:
I see your point, TaskDefinition doesn't have a label but looks like the registeredAppName should be the label, see here: https://github.com/spring-cloud/spring-cloud-dataflow/blob/28e62fd2b6c20202331c8448d75b3f4e50789345/spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/DataFlowAppDefinition.java#L62